src/client/components/pages/parts/search-results.js

```diff
@@ -21,122 +21,257 @@ import * as bootstrap from 'react-bootstrap';
+			return (
+				<tr key={result.bbid}>
+					{
+						!this.props.condensed &&
+						<td>
+							{
+								this.props.user ?
+									<input
+										checked={this.state.selected.find(selected => selected.bbid === result.bbid)}
+										className="checkboxes"
+										type="checkbox"
+										onChange={() => this.toggleRow(result)}
+									/> : null
+							}
+							{genEntityIconHTMLElement(result.type)}{_startCase(result.type)}
+						</td>
+					}
 					<td>
-						{genEntityIconHTMLElement(result.type)}{_startCase(result.type)}
+						<a href={link}>
+							{name} {disambiguation}
+						</a>
 					</td>
+					{
+						!this.props.condensed &&
+						<td>
+							{secondaryAliases}
+						</td>
+					}
+				</tr>
+			);
+		});
+		let tableCssClasses = 'table table-striped';
+		if (this.props.condensed) {
+			tableCssClasses += ' table-condensed';
+		}
+
+		return (
+			<div>
+				{
+					this.props.user ?
+						<div>
+							<AddToCollectionModal
+								bbids={this.state.selected.map(selected => selected.bbid)}
+								closeModalAndShowMessage={this.closeModalAndShowMessage}
+								entityType={this.state.selected[0]?.type}
+								handleCloseModal={this.onCloseModal}
+								show={this.state.showModal}
+								userId={this.props.user.id}
+							/>
+						</div> : null
 				}
-				<td>
-					<a href={link}>
-						{name} {disambiguation}
-					</a>
-				</td>
 				{
-					!props.condensed &&
-					<td>
-						{secondaryAliases}
-					</td>
+					!this.props.condensed &&
+					<h3 style={{color: '#754e37'}}>
+						Search Results
+					</h3>
 				}
-			</tr>
-		);
-	});
-	let tableCssClasses = 'table table-striped';
-	if (props.condensed) {
-		tableCssClasses += ' table-condensed';
-	}
-	return (
-		<div>
-			{
-				!props.condensed &&
-				<h3 style={{color: '#754e37'}}>
-					Search Results
-				</h3>
-			}
-			<hr className="thin"/>
-			<Table
-				responsive
-				className={tableCssClasses}
-			>
+				<hr className="thin"/>
+				<Table
+					responsive
+					className={tableCssClasses}
+				>
+					{
+						!this.props.condensed &&
+						<thead>
+							<tr>
+								<th className="col-sm-3">Type</th>
+								<th className="col-sm-5">Name</th>
+								<th className="col-sm-4">Aliases</th>
+							</tr>
+						</thead>
+					}
+					<tbody>
+						{results}
+					</tbody>
+					{
+						this.props.user ?
+							<Button
+								bsSize="small"
+								bsStyle="danger"
+								className="pull-left margin-top-d5"
+								type="button"
+								onClick={this.handleAddToCollection}
+							>
+								Add to Collection
+							</Button> : null
+					}
+				</Table>
 				{
-					!props.condensed &&
-					<thead>
-						<tr>
-							<th style={{width: '150px'}}>Type</th>
-							<th>Name</th>
-							<th>Aliases</th>
-						</tr>
-					</thead>
+					this.state.message.text ?
+						<Alert bsStyle={this.state.message.type} onDismiss={this.handleAlertDismiss}>{this.state.message.text}</Alert> : null
```

Comment:
This could probably benefit from a `className='margin-top-1'` to have some space between the button and the alert.